flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java

```diff
@@ -1294,9 +1304,118 @@ public void visit(LockStatementNode lockStatementNode) {
 
     @Override
     public void visit(ForkStatementNode forkStatementNode) {
-        handleDefaultStatementNode(forkStatementNode, () -> super.visit(forkStatementNode));
+        startNode(NodeKind.FORK, forkStatementNode);
+        forkStatementNode.namedWorkerDeclarations().forEach(this::visit);
+        endNode(forkStatementNode);
     }
 
+    @Override
+    public void visit(NamedWorkerDeclarator namedWorkerDeclarator) {
+        // Analyze the worker init statements
+        namedWorkerDeclarator.workerInitStatements().forEach(statement -> statement.accept(this));
+
+        // Generate a parallel flow node for the named workers
+        startNode(NodeKind.PARALLEL_FLOW);
+        NodeList<NamedWorkerDeclarationNode> workers = namedWorkerDeclarator.namedWorkerDeclarations();
+        LineRange startLineRange = workers.get(0).lineRange();
```

Comment:
Lets handle the node list being empty case.

Reply:
Is it possible for a namedWorkerDeclarator node to be constructed without at least one namedWorkerDeclaration?

Reply:
If it is no, fine.